### DISCUSSION TITLE
FNIRSI GC-01: Volunteers needed for bootloader dump!

### DISCUSSION THREAD

**Discussion by Gissio** [Gissio/radpro#59] [Announcements]
I have a request for people who own a GC-01 with a CH32F103R8T6 (WCH) processor.

In order to help people who are having trouble installing Rad Pro on said device, I need volunteers to do the following:

* Install this special firmware on your device with the bootloader's USB drive:

[fnirsi-gc01-bootloader-dumper.zip](https://github.com/Gissio/radpro/files/14898298/fnirsi-gc01-bootloader-dumper.zip)

* Restart your GC-01.
* A USB serial port device should appear on your computer (you might need to unplug/replug the device). Under Windows, open the Device Manager, go to the "Ports (COM & LPT) section and take note of the COM port. Next, download PuTTY (https://the.earth.li/~sgtatham/putty/latest/w64/putty.exe), start it, click on "Serial" and type the COM port in "Serial line". A terminal window should show up. Press ENTER and the memory dump should start. Next, go to the window menu (upper left), select "Copy All to Clipboard" and paste in a .txt file. Attach the file to a message in this thread.
* Reload the original firmware or Rad Pro.

If somebody else already uploaded the dump, it would be interesting that you repeat the process, just in case there are different versions of the bootloader.

**Comment by alfmck** [comment]


**Reply by Gissio** [reply]
Greatly appreciated. Just to confirm: is this from GC-01 with a WCH processor?

**Comment by alfmck** [comment]


**Reply by alfmck** [reply]
Full file
[downloader dump.txt](https://github.com/Gissio/radpro/files/14921881/downloader.dump.txt)

**Comment by alfmck** [comment]
Ch32f103r8t6Sent from my iPhoneOn 9 Apr 2024, at 21:04, Gissio ***@***.***> wrote:﻿
Greatly appreciated. Just to confirm: is this from GC-01 with a WCH processor?

**Comment by PsychoBitchy** [comment]
Hello, here is mine.

[FNIRSI-GC-01_Bootloader.txt](https://github.com/Gissio/radpro/files/15197208/FNIRSI-GC-01_Bootloader.txt)

**Reply by Gissio** [reply]
Greatly appreciated, thanks a lot!

**Comment by Captainslarty** [comment]
Here is mine for the GC-01 with apm32f103rb
[Joes Blue GC-01 Bootloader.zip](https://github.com/Gissio/radpro/files/15258174/Joes.Blue.GC-01.Bootloader.zip)
Cheers.
Joe

some images of my unit. *Banggood*
![20240508_073753](https://github.com/Gissio/radpro/assets/34556125/97e93eda-6770-4dd4-9281-ab84919e7774)
![20240508_073804](https://github.com/Gissio/radpro/assets/34556125/cd920493-73d8-47bd-bcb5-3f80bf23833b)
![20240508_074149](https://github.com/Gissio/radpro/assets/34556125/68177f24-a67c-4c20-aa67-bef50b2d9847)
![20240508_074217](https://github.com/Gissio/radpro/assets/34556125/7c55e3c6-b0ce-427d-82b1-e3a4be88056d)
![20240508_074226](https://github.com/Gissio/radpro/assets/34556125/aa88816c-af31-497d-b396-2d8ba550a341)
![20240508_074238](https://github.com/Gissio/radpro/assets/34556125/084d3b69-de15-4447-8ce5-da69cef2d55a)
![20240508_074342](https://github.com/Gissio/radpro/assets/34556125/ba8043f0-c812-436e-8fb6-3ebea7deaa72)
![20240508_074550](https://github.com/Gissio/radpro/assets/34556125/84224954-7fb3-4d44-9a35-7221df7f387c)
![20240508_074603](https://github.com/Gissio/radpro/assets/34556125/24dd0aaa-a46b-45c6-9ad9-5393771ee731)

**Reply by dc1rdb** [reply]
Hi Gissio, how can Captainslarty's bootloader be converted into a ST-Link loadable bin? I would like to try if it fixes my bootloader problem, since his hardware looks identical to mine...
Thanks :)

**Reply by Gissio** [reply]
Unfortunately Captainslarty's bootloader is the same as the one bundled with the flashtool.

**Reply by dc1rdb** [reply]
Understood. Thanks for the heads up.

**Comment by Jozer99** [comment]
Here is the firmware from my device, as requested.  
[APM32_bootloader.txt](https://github.com/user-attachments/files/24419721/APM32_bootloader.txt)
